django/utils/dateparse.py

```diff
@@ -136,7 +137,10 @@ def parse_duration(value):
         match = iso8601_duration_re.match(value) or postgres_interval_re.match(value)
     if match:
         kw = match.groupdict()
-        days = datetime.timedelta(float(kw.pop('days', 0) or 0))
+        days = float(kw.pop('days', 0) or 0)
+        if not timedelta.min.days <= days <= timedelta.max.days:
```

Comment:
Why is this needed? Isn't it reimplementing an exception raised by `timedelta()`? No tests seem to fail with it removed.